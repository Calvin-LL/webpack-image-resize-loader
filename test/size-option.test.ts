import webpack from "webpack";

import compile from "./helpers/compile";
import execute from "./helpers/execute";
import getCompiler from "./helpers/getCompiler";
import readAsset from "./helpers/readAsset";

describe('"size" option', () => {
  it("should work with { width: 10 }", async () => {
    const compiler = getCompiler({
      width: 10,
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it("should work with { width: 10000000 }", async () => {
    const compiler = getCompiler({
      width: 10000000,
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });
});
