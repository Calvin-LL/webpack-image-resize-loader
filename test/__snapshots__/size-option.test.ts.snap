// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"size" option should work with { width: 10 }: result 1`] = `"e9cac5dc1b3f4b537ff58229490ade0c.jpg"`;

exports[`"size" option should work with { width: 10000000 }: result 1`] = `"fdaf70a107ad3cebfaf1e03ff0601c1c.jpg"`;
